core/src/main/scala/org/apache/spark/FutureAction.scala

```diff
@@ -105,6 +108,7 @@ trait FutureAction[T] extends Future[T] {
  * A [[FutureAction]] holding the result of an action that triggers a single job. Examples include
  * count, collect, reduce.
  */
+@DeveloperApi
```

Comment:
Why are we exposing these classes as "stable API"? All of the public methods of `AsyncRDDFunctions` return `FutureAction`, not `SimpleFutureAction` or `ComplexFutureAction`. The fact that the latter two classes exist at all is an implementation detail.